benchmarks/libseff/parsing_dollars/main.c

```diff
@@ -0,0 +1,114 @@
+#include "seff.h"
+#include <stdio.h>
+
+DEFINE_EFFECT(read, 0, char, {});
+DEFINE_EFFECT(emit, 1, void, { int64_t value; });
+DEFINE_EFFECT(stop, 2, void, {});
+
+// Improves readability, but is not needed
+#define HANDLE(func, arg, handler) handler(seff_coroutine_new(func, arg))
+
+#define NEWLINE 10
+#define DOLLAR 36
+inline bool is_newline(char c) { return (int) c == NEWLINE; }
+inline bool is_dollar(char c) { return (int) c == DOLLAR; }
+
+static void* parse(void* parameter) {
+  int64_t a = 0;
+  while (true) {
+    char c = PERFORM(read);
+    if (is_dollar(c)) a++;
+    else if (is_newline(c)) {
+      PERFORM(emit, a);
+      a = 0;
+    }
+    else PERFORM(stop);
```

Comment:
Please double check, but I think that you may be leaking the lowermost coroutine (the one created on line 90) if this `PERFORM(stop)` is executed, since it's completely skipping the `feed` handler.

Reply:
You are right! I have fixed the issue by deleting the lowermost coroutine when STOP is performed (since we will not need it after that).

Reply:
Oh, I think you found (and fixed) another leak. The one I meant was, when you have three handlers installed (`sum` -> `catch` -> `feed`) and are executing the `parse` method, if you `PERFORM(stop)` you'll jump straight to the `catch` handler. This will free the coroutine created at line 99, then you return and delete (on line 45) the coroutine allocated at line 104.

I may be wrong on this, but I think the one allocated at line 93 is never released.

Reply:
Hmm I think you're right! In that case, since `feed` is the only method with access to the relevant coroutine, it seems that the only thing we can do is catch the `stop` effect inside feed, delete the coroutine, then call `PERFORM(stop)` again.

Can you think of a way that doesn't require this? It probably doesn't matter for performance, but it changes the semantics of the handlers.

Reply:
Yeah, it's not ideal, I think it may be ok to leak it for now.
We should think on how to fix this in the general case (this case is a bit easier, since it's the performing coroutine that needs freeing).
I guess it'd need to be something exception like (maybe that is caught at every handling point) and just releases and rethrows. This can be done either within libseff or externally.